frontend-v2/src/app/users/user-table.tsx

```diff
@@ -0,0 +1,287 @@
+      if (state === 'desc') return '▼'
+      return ''
+    }
+
+    return [
+      {
+        header: ({ column }) => (
+          <button
+            type="button"
+            onClick={column.getToggleSortingHandler()}
+            className="flex items-center gap-1"
+          >
+            <span>Name</span>
+            <span className="text-xs">
+              {sortIndicator(column.getIsSorted())}
+            </span>
+          </button>
+        ),
+        accessorKey: 'name',
+        cell: ({ row }) => (
+          <Link
+            href={`/users/${row.original.id}`}
+            className="font-semibold hover:underline"
+          >
+            {row.original.name}
+          </Link>
+        ),
+      },
+      {
+        header: ({ column }) => (
+          <button
+            type="button"
+            onClick={column.getToggleSortingHandler()}
+            className="flex items-center gap-1"
+          >
+            <span>Email</span>
+            <span className="text-xs">
+              {sortIndicator(column.getIsSorted())}
+            </span>
+          </button>
+        ),
+        accessorKey: 'email',
+        cell: ({ getValue }) => (
+          <span className="font-mono text-sm">{getValue<string>()}</span>
+        ),
+      },
+      {
+        id: 'chapter',
+        header: ({ column }) => (
+          <button
+            type="button"
+            onClick={column.getToggleSortingHandler()}
+            className="flex items-center gap-1"
+          >
+            <span>Chapter</span>
+            <span className="text-xs">
+              {sortIndicator(column.getIsSorted())}
+            </span>
+          </button>
+        ),
+        accessorFn: (row) => chapterMap.get(row.chapter_id) ?? '',
+        cell: ({ getValue }) => {
+          const chapterName = getValue<string>()
+          return chapterName || 'Unknown'
+        },
+      },
+      {
+        header: 'Roles',
+        accessorKey: 'roles',
+        cell: ({ getValue }) => {
+          const roles = getValue<string[]>()
+          if (!roles.length) return '—'
+          return roles.join(', ')
+        },
+      },
+      {
+        id: 'status',
+        header: ({ column }) => (
+          <button
+            type="button"
+            onClick={column.getToggleSortingHandler()}
+            className="flex items-center gap-1"
+          >
+            <span>Status</span>
+            <span className="text-xs">
+              {sortIndicator(column.getIsSorted())}
+            </span>
+          </button>
+        ),
+        accessorFn: (row) => (row.disabled ? 'Disabled' : 'Active'),
+        cell: ({ getValue, row }) => {
+          const status = getValue<string>()
+          const isDisabled = row.original.disabled
+          return (
+            <span
+              className={
+                isDisabled
+                  ? 'text-xs font-semibold text-destructive'
+                  : 'text-xs font-semibold text-emerald-700'
```

Comment:
fine as-is, but my default here would be to do something like:

```ts
clsx('text-xs font-semibold', isDisabled ? 'text-destructive' : 'text-emerald-700')
```

to factor out the common classnames